R/rw.R

```diff
@@ -0,0 +1,229 @@
+#'
+#' @examples
+#' # The empty cf object must be empty:
+#' is_empty.cf(cf())
+#'
+#' # The sample cf must not be empty:
+#' is_empty.cf(samplecf)
+is_empty.rw <- function (.rw) {
+  setequal(class(.rw), class(rw())) &&
+    is.null(names(.rw))
+}
+
+
+#' Arithmetically multiplies two reweighting factors 
+#'
+#' @param rw1,rw2 `rw_orig` object.
+#' @param nf1,nf2 Integer. Factors that determines the number of flavours you 
+#'        have for reweighting factor 1 and 2
+#'
+#' @return
+#' The value is
+#' \deqn{rw_1*nf1*rw_2*nf2) \,.}
+#'
+#' @export
+multiply.rw <- function(rw1, rw2, nf1 = 1, nf2 = 1) {
+
+  stopifnot(inherits(rw1, 'rw'))
+  stopifnot(inherits(rw2, 'rw'))
+ 
+  stopifnot(inherits(rw1, 'rw_orig'))
+  stopifnot(inherits(rw2, 'rw_orig'))
+
+  stopifnot(inherits(rw1, 'rw_meta'))
+  stopifnot(inherits(rw2, 'rw_meta'))
+ 
+  stopifnot(identical( rw1$conf.index, rw2$conf.index ))
+
+  rw <- rw()
+  rw$rw <- nf1*rw1$rw * nf2*rw2$rw
+  rw$max_value <- NA
+  rw$conf.index <- rw1$conf.index 
+
+  class(rw) <- append(class(rw), 'rw_orig')
+  class(rw) <- append(class(rw), 'rw_meta')
+  
+  return(rw)
+}
+
+#' Arithmetically multiply reweighting factors
+#'
+#' @param rw1,rw2 `rw_orig` objects.
+#'
+#' @export
+'*.cf' <- function (rw1, rw2) {
+  multiply.rw(cf1, cf2, nf1 = 1, nf2 = 1)
+}
+
+
+#' Checks whether an object is a rw
+#'
+#' @param x Object, possibly of class `rw`.
+#'
+#' @export
+is.rw <- function (x) {
+  inherits(x, "rw")
+}
+
+
+#' Plot a reweighting factor function
+#'
+#' @param x `rw` object
+#' @param neg.vec Numeric vector of length `rw$rw0`. This allows switching the
+#' sign for certain gauge configurations such that displaying in
+#' log-scale is sensible.
+#' @param rep See \code{\link{plotwitherror}}.
+#'
+#' @inheritParams plotwitherror
+#'
+#' @export
+plot.rw <- function(x, neg.vec = rep(1, times = length(rw$rw)), rep = FALSE, ...) {
+  rw <- x
+  stopifnot(inherits(rw, 'rw'))
+
+  val <- rw$rw/mean(rw$rw)
+  err <- rep(x=0,length(rw$rw))
+
+  df <- data.frame(t = c(1:length(val)),
+                   CF = val,
+                   Err = err)
+
+  plotwitherror(x = c(1:length(val)), y = neg.vec * df$CF, dy = df$Err, rep = rep, ...)
+
+  return(invisible(df))
+}
+
+#' Combine statistics for reweighting factor 
+#'
+#' @param rw1 `rw` object
+#' @param rw2 `rw` object
+#' @param reverse1 boolean object
+#' @param reverse1 boolean object
+#'
+#' @export
+addStat.rw <- function(rw1, rw2,reverse1=FALSE, reverse2=FALSE) {
+  stopifnot(inherits(rw1, 'rw'))
+  stopifnot(inherits(rw2, 'rw'))
+
+  if (is_empty.rw(rw1)) {
+    return (invisible(rw2))
+  }
+  if (is_empty.cf(rw2)) {
+    return (invisible(rw1))
+  }
+
+  stopifnot(inherits(rw1, 'rw_orig'))
+  stopifnot(inherits(rw2, 'rw_orig'))
+
+
+  stopifnot(inherits(rw1, 'rw_meta'))
+  stopifnot(inherits(rw2, 'rw_meta'))
+ 
+  stopifnot(inherits(rw1, 'rw_norm'))
+  stopifnot(inherits(rw2, 'rw_norm'))
+
+
+  #stopifnot(identical(rw1$conf.index == rw2$conf.index))
```

Comment:
remove commented out lines

Reply:
this I removed I think

Reply:
but it's right here

Reply:
it's still right there...

Reply:
I have not found this in origin/reweighting branch, it shows me also Outdated